plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools.tsx

```diff
@@ -0,0 +1,185 @@
+/**
+ * External dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { TabPanel, Button } from '@wordpress/components';
+import { recordEvent } from '@woocommerce/tracks';
+import { Pill, EmptyContent, Spinner } from '@woocommerce/components';
+import { flatMapDeep, uniqBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	CollapsibleCard,
+	CardDivider,
+	CardBody,
+	PluginCardBody,
+} from '~/marketing/components';
+import { STORE_KEY } from '~/marketing/data/constants';
+import { getInAppPurchaseUrl } from '~/lib/in-app-purchase';
+import './DiscoverTools.scss';
+
+const category = 'marketing';
+
+type SubcategoryType = {
+	slug: string;
+	name: string;
+};
+
+type TagType = {
+	slug: string;
+	name: string;
+};
+
+type Plugin = {
+	title: string;
+	description: string;
+	url: string;
+	icon: string;
+	product: string;
+	plugin: string;
+	categories: Array< string >;
+	subcategories: Array< SubcategoryType >;
+	tags?: Array< TagType >;
+};
+
+type SelectResult = {
+	isLoading: boolean;
+	plugins: Plugin[];
+};
+
+/**
+ * Return tabs (`{ name, title }`) for the TabPanel.
+ *
+ * Subcategories that have no plugins
+ * will not be displayed as a tab in the UI.
+ * This is done by doing the following:
+ *
+ * 1. Get an array of unique subcategories from the list of plugins.
+ * 2. Map the subcategories schema into tabs schema.
+ */
+const getTabs = ( plugins: Plugin[] ) => {
+	const pluginSubcategories = uniqBy(
+		flatMapDeep( plugins, ( p ) => p.subcategories ),
+		( subcategory ) => subcategory.slug
+	);
+
+	return pluginSubcategories.map( ( subcategory ) => ( {
+		name: subcategory.slug,
+		title: subcategory.name,
+	} ) );
+};
+
+const renderPluginCardBodies = ( plugins: Plugin[] ) => {
+	return plugins.map( ( el, idx ) => {
+		return (
+			<Fragment key={ el.product }>
+				<PluginCardBody
+					icon={ <img src={ el.icon } alt={ el.title } /> }
+					name={ el.title }
+					pills={ el.tags?.map( ( tag ) => (
+						<Pill key={ tag.slug }>{ tag.name }</Pill>
+					) ) }
+					description={ el.description }
+					button={
+						<Button
+							variant="secondary"
+							href={ getInAppPurchaseUrl( el.url ) }
+							onClick={ () => {
+								recordEvent(
+									'marketing_recommended_extension',
+									{ name: el.title }
+								);
+							} }
+						>
+							{ __( 'Get started', 'woocommerce' ) }
+						</Button>
+					}
+				/>
+				{ idx !== plugins.length - 1 && <CardDivider /> }
+			</Fragment>
+		);
+	} );
+};
+
+export const DiscoverTools = () => {
+	const { isLoading, plugins } = useSelect< SelectResult >(
+		( select ) => {
+			const { getRecommendedPlugins, isResolving } = select( STORE_KEY );
+
+			return {
+				isLoading: isResolving( 'getRecommendedPlugins', [ category ] ),
+				plugins: getRecommendedPlugins( category ),
+			};
+		},
+		[ category ]
+	);
+
+	/**
+	 * Renders card body.
```

Comment:
I would suggest adding some unit tests for this component to verify the logic, but it's okay to add them in another PR.

Reply:
@chihsuan , thanks, I added some tests for the component in [`0cb4fee` (#34415)](https://github.com/woocommerce/woocommerce/pull/34415/commits/0cb4fee7e1cddaa54a35beb6eeaf8eae97815c67).